stl/inc/format

```diff
@@ -2074,6 +2074,145 @@ wstring format(const locale& _Loc, wstring_view _Fmt, const _Types&... _Args) {
+            *_Out++ = _Ch;
+        }
+        ++_Count;
+        return *this;
+    }
+
+    _Format_to_n_iterator& operator*() {
+        return *this;
+    }
+
+    _Format_to_n_iterator& operator++() {
+        return *this;
+    }
+
+    _Format_to_n_iterator& operator++(int) {
+        return *this;
+    }
+};
+
+template <class _OutputIt>
+struct format_to_n_result {
+    _OutputIt out;
+    iter_difference_t<_OutputIt> size;
+};
+
+template <class _OutputIt, class... _Types>
+format_to_n_result<_OutputIt> format_to_n(
+    _OutputIt _Out, iter_difference_t<_OutputIt> _Max, string_view _Fmt, const _Types&... _Args) {
+    _Format_to_n_iterator<_OutputIt, char> _It{._Out = _Out, ._Max = _Max};
+
+    using _Context  = basic_format_context<decltype(_It), char>;
+    auto _Arg_store = _Format_arg_store<_Context, _Types...>{_Args...};
+    _It             = _STD vformat_to(_It, _Fmt, _STD move(_Arg_store));
+    return {.out = _It._Out, .size = _It._Count};
+}
+
+template <class _OutputIt, class... _Types>
+format_to_n_result<_OutputIt> format_to_n(
+    _OutputIt _Out, iter_difference_t<_OutputIt> _Max, wstring_view _Fmt, const _Types&... _Args) {
+    _Format_to_n_iterator<_OutputIt, wchar_t> _It{._Out = _Out, ._Max = _Max};
+
+    using _Context  = basic_format_context<decltype(_It), wchar_t>;
+    auto _Arg_store = _Format_arg_store<_Context, _Types...>{_Args...};
+    _It             = _STD vformat_to(_It, _Fmt, _STD move(_Arg_store));
+    return {.out = _It._Out, .size = _It._Count};
+}
+
+template <class _OutputIt, class... _Types>
+format_to_n_result<_OutputIt> format_to_n(
+    _OutputIt _Out, iter_difference_t<_OutputIt> _Max, const locale& _Loc, string_view _Fmt, const _Types&... _Args) {
+    _Format_to_n_iterator<_OutputIt, char> _It{._Out = _Out, ._Max = _Max};
+
+    using _Context  = basic_format_context<decltype(_It), char>;
+    auto _Arg_store = _Format_arg_store<_Context, _Types...>{_Args...};
+    _It             = _STD vformat_to(_It, _Loc, _Fmt, _STD move(_Arg_store));
+    return {.out = _It._Out, .size = _It._Count};
+}
+
+template <class _OutputIt, class... _Types>
+format_to_n_result<_OutputIt> format_to_n(
+    _OutputIt _Out, iter_difference_t<_OutputIt> _Max, const locale& _Loc, wstring_view _Fmt, const _Types&... _Args) {
+    _Format_to_n_iterator<_OutputIt, wchar_t> _It{._Out = _Out, ._Max = _Max};
+
+    using _Context  = basic_format_context<decltype(_It), wchar_t>;
+    auto _Arg_store = _Format_arg_store<_Context, _Types...>{_Args...};
+    _It             = _STD vformat_to(_It, _Loc, _Fmt, _STD move(_Arg_store));
+    return {.out = _It._Out, .size = _It._Count};
+}
+
+template <class _CharT>
+struct _Counting_iterator {
+    size_t _Count = 0;
+
+    using iterator_category = output_iterator_tag;
+    using value_type        = void;
+    using difference_type   = ptrdiff_t;
+    using pointer           = void;
+    using reference         = void;
+
+    _Counting_iterator& operator=(_CharT) {

```

Comment:
If this assignment operator were a template, the class wouldn't need to be a template, saving several microseconds of compile time.